mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala

```diff
@@ -439,6 +427,7 @@ class MesosClientImpl(
     }
 
   private val callSerializer: Flow[Call, (Array[Byte], Call.Type), NotUsed] = Flow[Call]
+    .map(call => call.toBuilder.setFrameworkId(frameworkId).build())
```

Comment:
I think this is a little more elegant than forcing users to use the `MesosCalls` factory.

Reply:
hmm, also potentially inefficient? It is nice to keep the decoupling of setting the FrameworkID from the scheduler

Reply:
But then we cannot restart the connection. At least I could not find a design that allowed to do so.